plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java

```diff
@@ -1,140 +1,94 @@
 import com.mapbox.mapboxsdk.style.sources.Source;
 import com.mapbox.mapboxsdk.style.sources.VectorSource;
 
+import java.util.List;
 import java.util.Locale;
 
 import static com.mapbox.mapboxsdk.style.layers.PropertyFactory.textField;
 
-/**
- * The localization plugin enables localization of map labels into the user’s preferred language.
- * Preferred language is determined by the language that the device is set to in the device's settings.
- * <p>
- * Initialise this plugin in the {@link com.mapbox.mapboxsdk.maps.OnMapReadyCallback#onMapReady(MapboxMap)} and provide
- * a valid instance of a {@link MapboxMap}.
- * </p>
- */
 public final class LocalizationPlugin {
 
-  private MapboxMap mapboxMap;
-  private String selectedBackupLanguage;
+  private final MapboxMap mapboxMap;
 
-  /**
-   * Create a {@link LocalizationPlugin}.
-   *
-   * @param mapboxMap      the MapboxMap to apply the localization plugin to
-   * @param backupLanguage the default language for the map to fall back to if the device language is not
-   *                       supported by the Mapbox Streets style source.
-   *                       See https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview for a list of
-   *                       supported languages.
-   * @since 0.1.0
-   */
-  public LocalizationPlugin(@NonNull MapboxMap mapboxMap, @NonNull String backupLanguage) {
+  public LocalizationPlugin(@NonNull MapboxMap mapboxMap) {
     this.mapboxMap = mapboxMap;
-    this.selectedBackupLanguage = backupLanguage;
-    setMapTextLanguage(getDeviceLanguage());
   }
 
-  /**
-   * Retrieve the code for the language that the device is currently set to.
-   *
-   * @return The correct code according to ISO-639-1
+  /*
+   * Map languages
    */
-  public String getDeviceLanguage() {
-    if (Build.VERSION.SDK_INT >= 21) {
-      String languageTag = Locale.getDefault().toLanguageTag();
-      if (languageTag.contains("Hans")) {
-        return "zh-Hans";
-      } else if (languageTag.contains("Hant")) {
-        return "zh";
-      } else {
-        return Locale.getDefault().toLanguageTag().substring(0, 2);
-      }
-    } else {
-      if (Locale.getDefault().getLanguage().equalsIgnoreCase("IW")) {
-        return "he";
-      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("IN")) {
-        return "id";
-      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("JI")) {
-        return "yi";
-      } else if (Locale.getDefault().getLanguage().equalsIgnoreCase("JW")) {
-        return "jv";
-      } else if (Locale.getDefault().getLanguage().contains("zh-Hans")
-        || Locale.getDefault().getDisplayName().contains("简体")) {
-        return "zh-Hans";
-      } else if (Locale.getDefault().getLanguage().contains("zh-Hant")
-        || Locale.getDefault().getDisplayName().contains("繁體")) {
-        return "zh";
-      } else {
-        return Locale.getDefault().getLanguage();
-      }
-    }
+
+  public void setMapLanguage() {
```

Comment:
without looking at the implementation it's not clear from the method name / parameters that this sets the default locale.